packages/gatsby-plugin-utils/src/validate.ts

```diff
@@ -11,19 +11,19 @@ interface IOptions {
   validateExternalRules?: boolean
 }
 
-export async function validateOptionsSchema<PluginOptions = object>(
+export async function validateOptionsSchema(
   pluginSchema: ObjectSchema,
-  pluginOptions: PluginOptions,
-  options: IOptions = {}
-): Promise<PluginOptions> {
-  if (options.validateExternalRules === false) {
-    const result = pluginSchema.validate(pluginOptions, {
-      ...validationOptions,
-      externals: false,
-    })
-    if (result.error) throw result.error
-    return result.value
+  pluginOptions: IPluginInfoOptions,
+  options: IOptions = {
+    validateExternalRules: true,
   }
+): Promise<IPluginInfoOptions> {
+  const { validateExternalRules } = options
 
-  return pluginSchema.validateAsync(pluginOptions, validationOptions)
+  const { value } = await pluginSchema.validateAsync(pluginOptions, {
+    ...validationOptions,
+    externals: validateExternalRules,
+  })
+
+  return value
 }
```

Comment:
I didn't understand why we were running sync and adding logic when we could always do async and pass in the `validateExternalRules`...let me know if there was reasoning and I can change it back.